external/wakaama/core/data.c

```diff
@@ -439,15 +440,27 @@ int lwm2m_data_parse(lwm2m_uri_t * uriP,
         if (*dataP == NULL) return 0;
         (*dataP)->id = uriP->resourceId;
         (*dataP)->type = LWM2M_TYPE_STRING;
-        return prv_setBuffer(*dataP, buffer, bufferLen);
+        res = prv_setBuffer(*dataP, buffer, bufferLen);
```

Comment:
Dear. dr-venkman
I have checked the wakaama in eclipse and applied some lines fix the memory leak problem.
I will modify it to the previous code as you said.
Thanks.